compiler/rustc_lint/src/context.rs

```diff
@@ -766,7 +766,39 @@ pub trait LintContext: Sized {
                 BuiltinLintDiagnostics::NamedAsmLabel(help) => {
                     db.help(&help);
                     db.note("see the asm section of Rust By Example <https://doc.rust-lang.org/nightly/rust-by-example/unsafe/asm.html#labels> for more information");
-                }
+                },
+                BuiltinLintDiagnostics::UnexpectedCfg(span, name, value) => {
+                    let mut possibilities: Vec<Symbol> = if value.is_some() {
+                        let Some(values_valid) = &sess.parse_sess.check_config.values_valid else {
+                            bug!("it shouldn't be possible to have a diagnostic on a value if values checking is not enable");
+                        };
+                        let Some(values) = values_valid.get(&name) else {
+                            bug!("it shouldn't be possible to have a diagnostic on a value whose name is not in values");
+                        };
+                        values.iter().map(|&s| s).collect()
+                    } else {
+                        let Some(names_valid) = &sess.parse_sess.check_config.names_valid else {
+                            bug!("it shouldn't be possible to have a diagnostic on a value if values checking is not enable");
+                        };
+                        names_valid.iter().map(|s| *s).collect()
+                    };
+
+                    // Show the full list if all possible values for a given name, but don't do it
+                    // for names as the possibilities could be very long
+                    if value.is_some() && !possibilities.is_empty() {
+                        // Sorting can take some time, so we only do it if required
+                        possibilities.sort();
```

Comment:
Sorting `Symbol`s is not very reproducible, it's better to sort their string representations.

Reply:
I wanted to avoid creating another `Vec` but you're right it's better and safer.